src/controllers/mainController.ts

```diff
@@ -252,6 +258,45 @@ export default class MainController implements vscode.Disposable {
         });
     }
 
+    /**
+     * Helper to script a node based on the script operation
+     */
+    public async scriptNode(node: TreeNodeInfo, operation: ScriptOperation, executeScript: boolean = false): Promise<void> {
+        let actionPromise = new Promise<boolean>(async (resolve, reject) => {
+            const nodeUri = ObjectExplorerUtils.getNodeUri(node);
+            let connectionCreds = Object.assign({}, node.connectionCredentials);
+            const databaseName = ObjectExplorerUtils.getDatabaseName(node);
+            // if not connected or different database
+            if (!this.connectionManager.isConnected(nodeUri) ||
+                connectionCreds.database !== databaseName) {
+                // make a new connection
+                connectionCreds.database = databaseName;
+                if (!this.connectionManager.isConnecting(nodeUri)) {
+                    const promise = new Deferred<boolean>();
+                    await this.connectionManager.connect(nodeUri, connectionCreds, promise);
+                    await promise;
+                }
+            }
+            const selectStatement = await this._scriptingService.script(node, nodeUri, operation);
+            const editor = await this._untitledSqlDocumentService.newQuery(selectStatement);
+            let uri = editor.document.uri.toString();
+            let title = path.basename(editor.document.fileName);
+            const queryUriPromise = new Deferred<boolean>();
+            await this.connectionManager.connect(uri, connectionCreds, queryUriPromise);
+            await queryUriPromise;
+            this._statusview.languageFlavorChanged(uri, Constants.mssqlProviderName);
+            this._statusview.sqlCmdModeChanged(uri, false);
+            if (executeScript) {
+                const queryPromise = new Deferred<boolean>();
+                await this._outputContentProvider.runQuery(this._statusview, uri, undefined, title, queryPromise);
+                await queryPromise;
+                await this.connectionManager.connectionStore.removeRecentlyUsed(<IConnectionProfile>connectionCreds);
```

Comment:
Why not just have runQuery take in a param that tells it whether to add it to recently used? That's what we do in ADS and it makes it much better than having to remember to remove it each time you didn't want to save it (plus gets rid of any possible state issues where things query for connections between when it's added and when it's removed)